recipes/pytrilinos/build.sh

```diff
@@ -0,0 +1,49 @@
+mkdir -p build
+cd build
+
+if [ $(uname) == Darwin ]; then
+    export CXXFLAGS="-stdlib=libc++"
+fi
+
+cmake \
+  -D CMAKE_BUILD_TYPE:STRING=RELEASE \
+  -D CMAKE_INSTALL_PREFIX:PATH=$PREFIX \
+  -D BUILD_SHARED_LIBS:BOOL=ON \
+  -D TPL_ENABLE_MPI:BOOL=ON \
+  -D MPI_BASE_DIR:PATH=$PREFIX \
+  -D MPI_EXEC:FILEPATH=$PREFIX/bin/mpiexec \
```

Comment:
Tests fail with the message below because we are running as root in the docker container,
```
--------------------------------------------------------------------------
mpiexec has detected an attempt to run as root.
Running at root is *strongly* discouraged as any mistake (e.g., in
defining TMPDIR) or bug can result in catastrophic damage to the OS
file system, leaving your system in an unusable state.

You can override this protection by adding the --allow-run-as-root
option to your cmd line. However, we reiterate our strong advice
against doing so - please do so at your own risk.
--------------------------------------------------------------------------
```